http/server_test.ts

```diff
@@ -1130,6 +1130,50 @@ Deno.test("Server should not reject when the handler throws", async () => {
   await servePromise;
 });
 
+Deno.test("Server should not close the downstream connection when the response stream throws", async () => {
+  const listenOptions = {
+    hostname: "localhost",
+    port: 4505,
+  };
+  const listener = Deno.listen(listenOptions);
+
+  const postRespondWith = deferred();
+  let n = 0;
+
+  const handler = () => {
+    return new Response(
+      new ReadableStream({
+        async start() {
+          await delay(0);
+          try {
+            throw new Error("test-error");
+          } finally {
+            n++;
+            postRespondWith.resolve();
```

Comment:
It's a little concerning that this deferred is resolved multiple times. It can only be resolved once tho.

Reply:
We could remove the deferred. After all, unless somebody has a better idea, this test needs to rely on `await delay(10)` for the following 2 requests (as explained in the code). We might as well use the same pattern for the first request, and at least be consistent.

Reply:
Maybe just create a new deferred on the same variable on each request? Then you don't need the `delay`

Reply:
this is what I mean in the code comment. If I do that and the test fails, the second `await postRespondWith;` hangs forever, cause the connection got broken and the `handler` is not executed again

Reply:
nvm, this test is wrong, and passes because there's a bug in Deno (broken streams should actually close the underlying connection in http1). Going to rewrite the test with http2 with a pattern suggested by @lucacasonato (thanks!)